src/axom/quest/examples/quest_shape_in_memory.cpp

```diff
@@ -929,80 +1000,155 @@ axom::sidre::View* getElementVolumes(
+          volView[cellIdx] = quadsView[cellIdx].area();
+        });
     }
+  }  // end of case 2
+  break;
+  case 3:
+  {
+    using HexahedronType = axom::primal::Hexahedron<double, 3>;
 
-    // Set vertex coords to zero if within threshold.
-    // (I don't know why we do this.  I'm following examples.)
-    axom::ArrayView<double> flatCoordsView(
-      (double*)vertCoords.data(),
-      vertCoords.size() * Point3D::dimension());
-    assert(flatCoordsView.size() == cellCount * NUM_VERTS_PER_HEX * 3);
-    axom::for_all<ExecSpace>(
-      cellCount * 3,
-      AXOM_LAMBDA(axom::IndexType i) {
-        if(axom::utilities::isNearlyEqual(flatCoordsView[i], 0.0, ZERO_THRESHOLD))
-        {
-          flatCoordsView[i] = 0.0;
-        }
-      });
+    if(volSidreView == nullptr)
+    {
+      mfem::Mesh* mesh = dc->GetMesh();
 
-    // Initialize hexahedral elements.
-    axom::Array<HexahedronType> hexes(cellCount, cellCount);
-    auto hexesView = hexes.view();
-    axom::for_all<ExecSpace>(
-      cellCount,
-      AXOM_LAMBDA(axom::IndexType cellIdx) {
-        // Set each hexahedral element vertices
-        hexesView[cellIdx] = HexahedronType();
+      constexpr int NUM_VERTS_PER_HEX = 8;
+      constexpr int NUM_COMPS_PER_VERT = 3;
+      constexpr double ZERO_THRESHOLD = 1.e-10;
+
+      axom::Array<Point3D> vertCoords(cellCount * NUM_VERTS_PER_HEX,
+                                      cellCount * NUM_VERTS_PER_HEX);
+      auto vertCoordsView = vertCoords.view();
+
+      // This runs only only on host, because the mfem::Mesh only uses host memory, I think.
+      for(axom::IndexType cellIdx = 0; cellIdx < cellCount; ++cellIdx)
+      {
+        // Get the indices of this element's vertices
+        mfem::Array<int> verts;
+        mesh->GetElementVertices(cellIdx, verts);
+        SLIC_ASSERT(verts.Size() == NUM_VERTS_PER_HEX);
+
+        // Get the coordinates for the vertices
         for(int j = 0; j < NUM_VERTS_PER_HEX; ++j)
         {
-          int vertIndex = (cellIdx * NUM_VERTS_PER_HEX) + j;
-          auto& hex = hexesView[cellIdx];
-          hex[j] = vertCoordsView[vertIndex];
+          int vertIdx = cellIdx * NUM_VERTS_PER_HEX + j;
+          for(int k = 0; k < NUM_COMPS_PER_VERT; k++)
+          {
+            vertCoordsView[vertIdx][k] = (mesh->GetVertex(verts[j]))[k];
+          }
         }
-      });  // end of loop to initialize hexahedral elements and bounding boxes
+      }
 
-    // Allocate and populate cell volumes.
-    volSidreView = dc->AllocNamedBuffer(volFieldName, cellCount);
-    axom::ArrayView<double> volView(volSidreView->getData(),
-                                    volSidreView->getNumElements());
-    axom::for_all<ExecSpace>(
-      cellCount,
-      AXOM_LAMBDA(axom::IndexType cellIdx) {
-        volView[cellIdx] = hexesView[cellIdx].volume();
-      });
+      // Set vertex coords to zero if within threshold.
+      // (I don't know why we do this.  I'm following examples.)
```

Comment:
It is sort of strange to only snap for coordinates near zero.
Anyone have thoughts/context?

We probably shouldn't check this comment in though :)

Reply:
> It is sort of strange to only snap for coordinates near zero.

I made a comment about it [here](https://teams.microsoft.com/l/message/19:f39b1bb4619947eb844797062b290238@thread.skype/1739299164714?tenantId=a722dec9-ae4e-4ae3-9d75-fd66e2680a63&groupId=761a2647-68cb-45ca-b03d-a1debb2011eb&parentMessageId=1739299164714&teamName=Axom%20Development&channelName=General&createdTime=1739299164714). 
I don't think these "snaps" are needed, and have removed them from `quest_shape_in_memory.cpp` and `IntersectionShaper.cpp`